sphinx/util/jsdump.py

```diff
@@ -1,194 +1,21 @@
-    i = 0
-    n = len(x)
-    stack: List[Union[List, Dict]] = []
-    obj: Any = nothing
-    key = False
-    keys = []
-    while i < n:
-        c = x[i]
-        if c == '{':
-            obj = {}
-            stack.append(obj)
-            key = True
-            keys.append(nothing)
-            i += 1
-        elif c == '[':
-            obj = []
-            stack.append(obj)
-            key = False
-            keys.append(nothing)
-            i += 1
-        elif c in '}]':
-            if key:
-                if keys[-1] is not nothing:
-                    raise ValueError("unfinished dict")
-                # empty dict
-                key = False
-            oldobj = stack.pop()
-            keys.pop()
-            if stack:
-                obj = stack[-1]
-                if isinstance(obj, dict):
-                    if keys[-1] is nothing:
-                        raise ValueError("invalid key object", oldobj)
-                    obj[keys[-1]] = oldobj
-                else:
-                    obj.append(oldobj)
-            else:
-                break
-            i += 1
-        elif c == ',':
-            if key:
-                raise ValueError("multiple keys")
-            if isinstance(obj, dict):
-                key = True
-            i += 1
-        elif c == ':':
-            if not isinstance(obj, dict):
-                raise ValueError("colon in list")
-            i += 1
-            if not key:
-                raise ValueError("multiple values")
-            key = False
-        else:
-            y: Any = None
-            m = _str_re.match(x, i)
-            if m:
-                y = decode_string(m.group()[1:-1])
-            else:
-                m = _int_re.match(x, i)
-                if m:
-                    y = int(m.group())
-                else:
-                    m = _name_re.match(x, i)
-                    if m:
-                        y = m.group()
-                        if y == 'true':
-                            y = True
-                        elif y == 'false':
-                            y = False
-                        elif y == 'null':
-                            y = None
-                        elif not key:
-                            raise ValueError("bareword as value")
-                    else:
-                        raise ValueError("read error at pos %d" % i)
-            i = m.end()
-            if isinstance(obj, dict):
-                if key:
-                    keys[-1] = y
-                else:
-                    obj[keys[-1]] = y
-                    key = False
-            else:
-                obj.append(y)
-    if obj is nothing:
-        raise ValueError("nothing loaded from string")
-    return obj
-
-
-def load(f: IO) -> Any:
-    return loads(f.read())
+"""This module implements a simple JavaScript serializer."""
+import json
+
+from sphinx.deprecation import RemovedInSphinx60Warning, deprecated_alias
+
+deprecated_alias(
```

Comment:
I think it's enough to add a `warnings.warn()` call to the top of the module without rewriting (replacing). Nobody harms by old implementation.